contracts/src/batch-bridge/L1BatchBridgeGateway.sol

```diff
@@ -0,0 +1,425 @@
+                (token, l2Token, cachedTokenState.pendingBatchIndex, cachedBatchState.hash)
+            ),
+            SAFE_BATCH_BRIDGE_GAS_LIMIT
+        );
+
+        emit BatchDeposit(_msgSender(), token, cachedTokenState.pendingBatchIndex, l2Token);
+
+        // update token state
+        unchecked {
+            cachedTokenState.pending -= uint128(cachedBatchState.amount);
+            cachedTokenState.pendingBatchIndex += 1;
+        }
+        tokens[token] = cachedTokenState;
+
+        // refund keeper fee
+        unchecked {
+            if (msg.value > depositFee + batchBridgeFee) {
+                _transferToken(address(0), _msgSender(), msg.value - depositFee - batchBridgeFee);
+            }
+        }
+    }
+
+    /**********************
+     * Internal Functions *
+     **********************/
+
+    /// @dev Internal function to deposit token.
+    /// @param token The address of token to deposit.
+    /// @param sender The address of token sender.
+    /// @param amount The amount of token to deposit.
+    function _deposit(
+        address token,
+        address sender,
+        uint96 amount
+    ) internal nonReentrant {
+        BatchConfig memory cachedBatchConfig = configs[token];
+        TokenState memory cachedTokenState = tokens[token];
+        _tryFinalzeCurrentBatch(token, cachedBatchConfig, cachedTokenState);
+
+        BatchState memory cachedBatchState = batches[token][cachedTokenState.currentBatchIndex];
+        if (amount < cachedBatchConfig.minAmountPerTx) {
+            revert ErrorDepositAmountTooSmall();
+        }
+
+        emit Deposit(sender, token, cachedTokenState.currentBatchIndex, amount, cachedBatchConfig.feeAmountPerTx);
+
+        // deduct fee and update cached state
+        unchecked {
+            amount -= cachedBatchConfig.feeAmountPerTx;
+            cachedTokenState.pending += amount;
+            cachedBatchState.amount += amount;
+            cachedBatchState.numDeposits += 1;
+        }
+
+        // compute the hash chain
+        bytes32 node = BatchBridgeCodec.encodeNode(sender, amount);
+        if (cachedBatchState.hash == bytes32(0)) {
+            bytes32 initialNode = BatchBridgeCodec.encodeInitialNode(token, cachedTokenState.currentBatchIndex);
+            // this is first tx in this batch
+            cachedBatchState.hash = BatchBridgeCodec.hash(initialNode, node);
+            cachedBatchState.firstDepositTimestamp = uint64(block.timestamp);
+        } else {
+            cachedBatchState.hash = BatchBridgeCodec.hash(cachedBatchState.hash, node);
+        }
+
+        batches[token][cachedTokenState.currentBatchIndex] = cachedBatchState;
+        tokens[token] = cachedTokenState;
+    }
+
+    /// @dev Internal function to finalze current batch.
+    ///      This function may change the value of `cachedTokenState`, which can be used in later operation.
+    /// @param token The address of token to finalze.
```

Comment:
```suggestion
    /// @param token The address of the finalization token
```

Reply:
`The address of the finalization token` doesn't really explain the meaning.

Reply:
ok, `finalze` typo --> `finalize`